scripts/bigquery_billing_report/bigquery_billing_report.py

```diff
@@ -147,64 +147,52 @@ def print_submission_pricing(ws_namespace, ws_name, query_sub_id, query_workflow
 
             if "calls" not in workflow_metadata_json:
-                # print ws_namespace, ws_name, submission_id, wf_id, workflow_metadata_json
-                continue
-
-            # workflow_metadata_json = workflow_id_to_metadata_json[wf_id]
-            calls_lower_json = dict(
-                (k.split(".")[-1].lower(), v) for k, v in workflow_metadata_json["calls"].iteritems())
-
-            if len(calls_lower_json) == 0:
                 print "\tNo Calls."
-            else:
-                calls_lower_translated_json = {}
-                for calls_name, call_json in calls_lower_json.iteritems():
-                    # from the Cromwell documentation call names can be translated into a different format
-                    # under certain circumstances.  We will try translating it here and see if we get a match.
-                    # Rules:
-                    # Any capital letters are lowercased.
-                    # Any character which is not one of [a-z], [0-9] or - will be replaced with -.
-                    # If the start character does not match [a-z] then prefix with x--
-                    # If the final character does not match [a-z0-9] then suffix with --x
-                    # If the string is too long, only take the first 30 and last 30 characters and add --- between them.
-                    # TODO: this is not a complete implementation - however I requested that cromwell metadata includes label
-                    # TODO: so that this translation is not necessary
-                    cromwell_translated_callname = re.sub("[^a-z0-9\-]", "-", call_name.lower())
-                    calls_lower_translated_json[cromwell_translated_callname] = call_json
-
-            print "|\t.--- Workflow %d of %d: %s (%s)" % (wf_count, wf_total_count, wf_id, workflow_json["status"])
-            wf_count += 1
-
+                continue
+                
             call_name_to_cost = defaultdict(list)
             for c in workflow_id_to_cost[wf_id]:
                 call_name_to_cost[c.call_name].append(c)
-
+            
             total = 0.0
             pd_cost = 0.0
             cpu_cost = 0.0
             other_cost = 0.0
-            for call_name in call_name_to_cost:
-                calls = calls_lower_json[call_name] if call_name in calls_lower_json else calls_lower_translated_json[
-                    call_name]
-
+            
+            for task_name, calls in workflow_metadata_json["calls"].iteritems():           
+                call_name = None
+                num_calls = 0
+                cache_hit = False
+                
+                # go through each call to find it's backendLabels to match up with the
+                # labels in BQ.  Calls that never ran will not have these, and will not be counted.
+                for call in calls:
+                    if "backendLabels" in call:
```

Comment:
There is some chance that the backendLabels don't get published if your workflow is very wide and metadata events were missed. Perhaps there can be another conditional for when backendLabels & callCaching keys are missing --> indicating something is severely wrong with this workflow and print a warning?

Reply:
added